TotalCrossVM/CMakeLists.txt

```diff
@@ -0,0 +1,620 @@
+cmake_minimum_required(VERSION 3.13.5)
+
+# set the project name
+project(TCVM VERSION 1.0)
+
+if(NOT DEFINED BUILD_SHARED_LIBS AND NOT DEFINED TCVM_SHARED)
+  if(CMAKE_GENERATOR STREQUAL Xcode)
+    set(TCVM_SHARED OFF) 
+    message("tcvm: ${TCVM_SHARED}")  
+  else() # others
+    set(TCVM_SHARED ON) 
+  endif()
+endif()
```

Comment:
Take a look at the code suggestion below, it seems to be easier to understand and also provides an explanation for why this is required.

```suggestion
# By default we build the TCVM as a shared library, except for Xcode
# which we do it as a static library.
if(NOT DEFINED TCVM_SHARED)
  if(CMAKE_GENERATOR STREQUAL Xcode)
    set(TCVM_SHARED OFF)
  else()
    set(TCVM_SHARED ON)
  endif()
endif()

message("tcvm shared library is: ${TCVM_SHARED}")  
```

Reply:
It will probably be one of the first changes after we get where we want (first merge with master).